src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -345,6 +345,14 @@ private SemanticRange DataToSemanticRange(int lineDelta, int charDelta, int leng
             return tokensResult;
         }
 
+        private static async Task<VersionStamp> GetDocumentSemanticVersionAsync(DocumentSnapshot documentSnapshot)
```

Comment:
I'd normally say this should take in a cancellation token and that token should be passed to `GetTextVersionAsync` below; however, we're awful and `GetTextVersionAsync` doesn't take in a cancellation token 😢

Reply:
I've been wondering how useful a "Just everything Async takes a CancellationToken" PR would be, I feel like we keep seeing spots of "oh, we're bad for not having it there".